lib/migrations/seed/Seeder.js

```diff
@@ -2,91 +2,59 @@
 // -------
 
 const path = require('path');
-const flatten = require('lodash/flatten');
-const extend = require('lodash/extend');
-const includes = require('lodash/includes');
-const { ensureDirectoryExists, getFilepathsInFolder } = require('../util/fs');
+const { ensureDirectoryExists } = require('../util/fs');
 const { writeJsFileUsingTemplate } = require('../util/template');
 const { yyyymmddhhmmss } = require('../util/timestamp');
-
-const filterByLoadExtensions = (extensions) => (value) => {
-  const extension = path.extname(value);
-  return includes(extensions, extension);
-};
+const { getMergedConfig } = require('./seeder-configuration-merger');
 
 // The new seeds we're performing, typically called from the `knex.seed`
 // interface on the main `knex` object. Passes the `knex` instance performing
 // the seeds.
 class Seeder {
   constructor(knex) {
     this.knex = knex;
-    this.config = this.setConfig(knex.client.config.seeds);
+    this.config = this.resolveConfig(knex.client.config.seeds);
   }
 
   // Runs seed files for the given environment.
   async run(config) {
-    this.config = this.setConfig(config);
-    let files = await this._listAll();
-    if (config && config.specific) {
-      files = files.filter((file) => path.basename(file) === config.specific);
-      if (files.length === 0) {
-        throw new Error(
-          `Invalid argument provided: the specific seed "${config.specific}" does not exist.`
-        );
-      }
-    }
+    this.config = this.resolveConfig(config);
+    const files = await this.config.seedSource.getSeeds(
```

Comment:
To reduce number of arguments, maybe pass only this.config ? we always call getSeeds with three arguments of a same config object.